### ISSUE TITLE
一些问题

### ISSUE THREAD

**Issue by lh34** [lyx3911/tianchi-ICDAR2023-ImageTamper#2]
task2任务，我想问一下，我使用的是官方的评估方法，为啥在本地计算出来的都为0，换了不同模型都是这样，这是为啥？

def iou(pred, mask, thres):
    mask = np.clip(mask, 0, 1)
    iou_m = []
    first_threshold = int(thres[0])
    num_thresholds = 256 - first_threshold
    iou_map = np.zeros(num_thresholds)
    for i in range(num_thresholds):
        tmp = np.zeros_like(pred)
        thre = i + first_threshold
        tmp[pred > thre] = 1
        iou_value = np.count_nonzero(np.logical_and(
            mask, tmp)) / np.count_nonzero(np.logical_or(mask, tmp))
        iou_map[i] = iou_value
    for item in thres:
        iou_m.append(np.max(iou_map[int(item)-first_threshold:]))
    return np.sum(iou_m)

def subprocess(thres, pred, mask):
    iou_score = iou(pred, mask, thres)
    return iou_score

def valid_one_epoch(model, valid_loader, CFG):
        model.eval()
        y_preds_iou = np.array([])
        pred_masks = []
        gt_masks = []
        num = 0  # 记录篡改图片的数量
        pbar = tqdm(enumerate(valid_loader), total=len(valid_loader), desc='Valid ')
        for _, (images, gt) in pbar:
        images  = images.to(CFG.device, dtype=torch.float) # [b, c, w, h]
        gt  = gt.to(CFG.device) 

        b = gt.size(0)  # batch
        y_preds = model(images) 
        y_preds  = torch.nn.Sigmoid()(y_preds) # [b, c, w, h]

        # 评估
        # 计算gt中篡改与未篡改图像
        gt_, _ = gt.reshape((b, -1)).max(dim=1)   # [b,]
        y_preds = (y_preds > CFG.thr).to(torch.float32)

        # 记录篡改图像的gt和pred的mask
        pred_mask = y_preds[gt_ == 1].cpu().detach().numpy()
        gt_mask = gt[gt_ == 1].cpu().detach().numpy()

        for i in range(pred_mask.shape[0]):
            pred_masks.append(pred_mask[i])
            gt_masks.append(gt_mask[i])

        y_preds, _ = y_preds.reshape((b,-1)).max(dim=1)
        y_preds = y_preds[gt_==0].cpu().detach().numpy()
        y_preds_iou = np.hstack((y_preds_iou, y_preds))

        # 记录gt中篡改图片的数量
        num += gt[gt_==1].size(0)

    thres = np.percentile(y_preds_iou, np.arange(91, 100, 1))
    scores = np.empty(num)
    with ProcessPoolExecutor() as ex:
        future_scores = [ex.submit(subprocess, thres, pred_masks[i], gt_masks[i]) for i in range(num)]
        for i, future in enumerate(as_completed(future_scores)):
            scores[i] = future.result()
    print("iou_score: {:.4f}".format(np.mean(scores)), flush=True)
    
    return np.mean(scores)

是不是我这个评估代码写的有问题？
